apps/meteor/ee/app/livechat-enterprise/server/hooks/resumeOnHold.ts

```diff
@@ -0,0 +1,73 @@
+import type { ILivechatVisitor, IMessage, IOmnichannelRoom, IRoom, IUser } from '@rocket.chat/core-typings';
+import { isEditedMessage, isOmnichannelRoom } from '@rocket.chat/core-typings';
+import { LivechatRooms, LivechatVisitors, Users } from '@rocket.chat/models';
+import { TAPi18n } from 'meteor/rocketchat:tap-i18n';
+import { OmnichannelEEService } from '@rocket.chat/core-services';
+
+import { callbacks } from '../../../../../lib/callbacks';
+import { callbackLogger } from '../../../../../app/livechat/server/lib/callbackLogger';
+
+const resumeOnHoldCommentAndUser = async (room: IOmnichannelRoom): Promise<{ comment: string; resumedBy: IUser }> => {
+	const {
+		v: { _id: visitorId },
+		_id: rid,
+	} = room;
+	const visitor = await LivechatVisitors.findOneById<Pick<ILivechatVisitor, 'name' | 'username'>>(visitorId, {
+		projection: { name: 1, username: 1 },
+	});
+	if (!visitor) {
+		callbackLogger.error(`[afterSaveMessage] Visitor Not found for room ${rid} while trying to resume on hold`);
+		throw new Error('Visitor not found while trying to resume on hold');
+	}
+
+	const guest = visitor.name || visitor.username;
+
+	const resumeChatComment = TAPi18n.__('Omnichannel_on_hold_chat_automatically', { guest });
+
+	const resumedBy = await Users.findOneById('rocket.cat');
+	if (!resumedBy) {
```

Comment:
can rocket.cat not exist?

Reply:
Then they are screwed :joy: A lot of our features like omnichannel-transcript, onhold, etc are built around using this `rocket.cat` user as a system user. So if some workspace doesn't have this user, then we should consider it as mis-configured and ask them to add this user back